gpjax/variational.py

```diff
@@ -0,0 +1,101 @@
+import abc
+from typing import Optional, Dict, Callable
+import jax.numpy as jnp
+from chex import dataclass
+
+import tensorflow_probability.substrates.jax.bijectors as tfb
+
+from .gps import Posterior
+from .types import Array, Dataset
+from .config import add_parameter, Identity, Softplus
+from .utils import I, concat_dictionaries
+
+import distrax
+
+Diagonal = distrax.Lambda(forward=lambda x: jnp.diagflat(x), inverse=lambda x: jnp.diagonal(x))
+
+FillDiagonal = distrax.Chain([Diagonal, Softplus])
+FillTriangular = distrax.Chain([tfb.FillTriangular()])
+
+
+@dataclass
+class VariationalFamily:
+    @property
+    @abc.abstractmethod
+    def params(self) -> Dict:
+        raise NotImplementedError
+
+
+@dataclass
+class VariationalGaussian(VariationalFamily):
+    inducing_inputs: Array
+    name: str = "Gaussian"
+    mu: Optional[Array] = None
+    sqrt: Optional[Array] = None
+    diag: Optional[bool] = False
+    whiten: Optional[bool] = True
+
+    def __post_init__(self):
+        """Initialize the variational family."""
+        self.num_inducing = self.inducing_inputs.shape[0]
+        add_parameter("inducing_inputs", Identity)
+
+        nz = self.num_inducing
+
+        if self.mu is None:
+            self.mu = jnp.zeros((nz, 1))
+            add_parameter("mu", Identity)
```

Comment:
Can we give this a more explicit name. `variational_mean` perhaps?

Reply:
Addressed in [be2eb5d](https://github.com/thomaspinder/GPJax/pull/43/commits/be2eb5da0b179efba6ddeee4c74ebe4232e56080).